llm/server.go

```diff
@@ -527,21 +539,20 @@ func (s *llmServer) getServerStatus(ctx context.Context) (ServerStatus, error) {
 		return ServerStatusError, fmt.Errorf("read health request: %w", err)
 	}
 
-	var status ServerStatusResp
+	var status ServerStatusResponse
 	if err := json.Unmarshal(body, &status); err != nil {
 		return ServerStatusError, fmt.Errorf("health unmarshal encode response: %w", err)
 	}
 
-	switch status.Status {
-	case "ok":
-		return ServerStatusReady, nil
-	case "no slot available":
-		return ServerStatusNoSlotsAvailable, nil
-	case "loading model":
+	result := NewServerStatus(status)
```

Comment:
I think it would be much cleaner if we can have a type be serialized and deserialized directly on the unmarshal instead of having to parse this status by matching a brittle string. Do you know if there is a reason we don't do that now?

Reply:
Yup! We had a bunch of roundabout logic that eventually arrived at a time.Duration:
https://github.com/ollama/ollama/pull/9324#discussion_r1968596091

Reply:
The existing interface was done that way to keep compatibility with llama.cpp's server.cpp. However, that is no longer needed so we can now shift to using native types.